components/Questionnaire/QuestionnaireActions.js

```diff
@@ -0,0 +1,60 @@
+import React from 'react'
+
+import { css, merge } from 'glamor'
+import { compose } from 'react-apollo'
+
+import { Button, A, InlineSpinner } from '@project-r/styleguide'
+
+import withT from '../../lib/withT'
+
+const styles = {
+  actions: css({
+    textAlign: 'center',
+    margin: '20px auto 20px auto'
+  }),
+  actionsLeft: css({
+    textAlign: 'left'
+  }),
+  reset: css({
+    textAlign: 'center',
+    marginTop: 10
+  }),
+  resetLeft: css({
+    display: 'inline-block',
+    textAlign: 'left',
+    lineHeight: '60px',
+    marginTop: 10,
+    marginLeft: 20
+  })
+}
+
+export default compose(withT)(
+  ({ t, onSubmit, onReset, updating, submitting, invalid, leftAlign }) => {
+    return (
+      <div {...merge(styles.actions, leftAlign && styles.actionsLeft)}>
+        <Button
+          primary
+          onClick={onSubmit}
+          disabled={updating || submitting || invalid}
+        >
+          {updating || submitting ? (
+            <InlineSpinner size={40} />
+          ) : (
+            t('questionnaire/submit')
+          )}
+        </Button>
+        {!!onReset && (
+          <div {...merge(styles.reset, leftAlign && styles.resetLeft)}>
+            {invalid ? (
+              t('questionnaire/invalid')
+            ) : (
+              <A href='#' onClick={onReset}>
```

Comment:
Prevent the hash from leaking into the url:
```suggestion
              <A href='#' onClick={(e) => {
                e.preventDefault()
                onReset()
              }}>
```

Reply:
Hey – I'll check a few times throughout the day. Shouldn't be a problem as long as the newsletter comes out during the night or there are a few hours of buffer between the reception of the texts and the publication of the newsletter (I need to do 1-2 things outside during the day).

Reply:
Thanks Anna. I'll ping you once it's ready.